### DISCUSSION TITLE
sort method does not support 'NULLS {FIRST|LAST}'

### DISCUSSION THREAD

**Discussion by jknisley** [jsforce/jsforce#1482] [Ideas]
When using ORDER BY in SOQL, null values are sorted first by default. However, SOQL lets you specify 'NULLS {FIRST|LAST}' as part of the ORDER BY (https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_orderby.htm).

Can SOQLBuilder.createOrderByClause() be updated to support sort params which include NULLS FIRST and NULLS LAST?

**Comment by vincekruger** [comment]
I would also love this support.

**Comment by Asafkbalink** [comment]
Me too. will V3 support it?

**Comment by paustint** [comment]
Shameless plug :D

I maintain a SOQL parsing/composing library that supports this - it isn't quite the same builder pattern as JSForce but it supports all SOQL features - https://github.com/jetstreamapp/soql-parser-js
